### PR TITLE
Turn off `combined.csv` file warnings for intermediate callbacks in watcher

### LINKED ISSUE #387: `combine_run_metrics` throws removed combined file warning on each FOV run through the watcher
**Is your feature request related to a problem? Please describe.**

Each time a new FOV is detected by the watcher and the MPH pipeline runs, the _combined.csv file gets removed and regenerated, but this also comes with a warning that indicates this happens. If the run contains a substantial amount of FOVs, this warning will pop up for each one except the first (with 50, this can get very wordy).

**Describe the solution you'd like**

There should at the very least be a flag that turns this off for the watcher. We could consider using `verbose` or something else.

Honestly, I wouldn't be against just removing the warning entirely either.

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Closes #387. The new watcher setup means we'll be defaulting to using intermediate callbacks for plotting QC and MPH metrics. This means `combined.csv` files, normally removed at the end of the watcher, now get removed after every FOV. The warning that gets thrown as a result can appear several times on cohorts with many FOVs, which we want to turn off in this situation.

**How did you implement your changes**

Pass a `warn_overwrite` flag to the relevant functions. Also include this check for QC combined file generation and overwriting.

**Remaining issues**

Need testing on a live run, per the usual.

### CODE REVIEW

File: src/toffy/watcher_callbacks.py [hunk 1]
```diff
@@ -358,16 +363,6 @@ def build_callbacks(
...
-    for run_cb in callbacks_with_prereq:
-        argnames = inspect.getfullargspec(getattr(RunCallbacks, run_cb))[0]
-        argnames = [argname for argname in argnames if argname != "self"]
-
-        misc_utils.verify_in_list(required_arguments=argnames, passed_arguments=list(kwargs.keys()))
-
-    callbacks_with_prereq = (
-        run_callbacks + intermediate_callbacks if intermediate_callbacks else run_callbacks[:]
-    )
-
```
Comment (camisowers) [inline src/toffy/watcher_callbacks.py:370]: Can you explain why this is this getting removed?

**Review by camisowers — CHANGES_REQUESTED** [review]
Looks good, just a question about the removed `build_callbacks` code.

File: src/toffy/watcher_callbacks.py (hunk 1 above)
Comment (alex-l-kong) [reply src/toffy/watcher_callbacks.py:370]: This particular section was duplicated, so it should be removed.

**Review by srivarra — APPROVED** [review]
Looks good!

**Review by camisowers — APPROVED** [review]

**Review by ngreenwald — APPROVED** [review]

**Pull request merged**